File: src/main/java/org/example/twopointers/WaterContainer.java
package org.example.twopointers;

public class WaterContainer {
    public int maxArea(int[] height) {
        int maxArea = 0;
        int l = 0, r = height.length-1;

        while (l < r) {
            int area = Math.min(height[l], height[r]) * (r - l);
            maxArea = Math.max(maxArea, area);

            if (height[l] < height[r]) l += 1;
            else if (height[l] > height[r]) r -= 1;
            else {
                if (height[l+1] < height[r-1]) l +=1;
                else if (height[l+1] > height[r-1]) r -= 1;
                else {
                    l += 1;
                }
            }
        }

        return maxArea;
    }
}

File: src/main/java/org/example/twopointers/WaterContainer.py
class Solution(object):
    def maxArea(self, height):
        l, r = 0, len(height) - 1
        maxArea = 0

        while l < r:
            if height[l] < height[r]:
                area = height[l] * (r - l)
                l += 1
            else:
                area = height[r] * (r - l)
                r -= 1
            if area > maxArea: maxArea = area
        return maxArea


obj = Solution()
# height = [1,8,6,2,5,4,8,3,7]
height = [1, 1]
print(obj.maxArea(height))